sdk/extensions/chronopost/karrio/providers/chronopost/error.py

```diff
@@ -0,0 +1,30 @@
+import typing
+from karrio.core.models import Message
+from karrio.providers.chronopost.utils import Settings
+from chronopost_lib import quickcostservice, shippingservice, trackingservice
+import karrio.lib as lib
+
+
+def parse_error_response(
+    response: lib.Element, settings: Settings, response_type: str
+) -> typing.List[Message]:
+    element = (
+        shippingservice.shippingMultiParcelV5Response
+        if response_type == "shippingMultiParcelV5Response"
+        else trackingservice.cancelSkybillResponse
+        if response_type == "cancelSkybillResponse"
+        else quickcostservice.calculateProductsResponse
+        if response_type == "calculateProductsResponse"
+        else trackingservice.trackSkybillV2Response
+    )
+    errors = lib.find_element(response_type, response, element)
+    return [
+        Message(
+            carrier_id=settings.carrier_id,
+            carrier_name=settings.carrier_name,
+            message=error.return_.errorMessage,
+            code=error.return_.errorCode,
+        )
+        for error in errors
+        if error.return_.errorCode != 0
+    ]
```

Comment:
This works. Here is a proposition. to make it lighter,
The good thing about the XML parsing library is that if it is fail-safe and will ignore nodes it doesn't know and process the one it recognizes.

So you don't need to know if it is a `shippingMultiParcelV5Response` or `calculateProductsResponse`. 
Either class will be able to parse all the responses and collect the errorCode and errorMessage since those are the fields they have in common.

Therefore, we just need to take one class that knows how to parse a `<return>...</return>` node and used it on all the responses to parse and check if they have an `errorCode` different from 0.

Please try this and let me know.

```suggestion
import typing
from karrio.core.models import Message
from karrio.providers.chronopost.utils import Settings
import chronopost_lib.shippingservice as ups
import karrio.lib as lib

ReturnType = ups. resultMultiParcelExpeditionValue

def parse_error_response(response: lib.Element, settings: Settings) -> typing.List[Message]:
    errors = lib.find_element("return", response, ReturnType)
    
    return [
        Message(
            carrier_id=settings.carrier_id,
            carrier_name=settings.carrier_name,
            message=error.errorMessage,
            code=error.errorCode,
        )
        for error in errors
        if error.errorCode != 0
    ]
```

Reply:
Mhm, this is definitely better!